spi/src/main/java/org/opensearch/jobscheduler/spi/JobDocVersion.java

```diff
@@ -24,6 +29,19 @@ public JobDocVersion(long primaryTerm, long seqNo, long version) {
         this.version = version;
     }
 
+    public JobDocVersion(StreamInput in) throws IOException {
```

Comment:
What is JobDocVersion, who uses this internally? 
Why should an extension care about this internal implementation of JS?

Reply:
The `JobDocVersion` represents the version of a particular job entry, which JobScheduler uses to determine whether or not a particular job entry is outdated, this can happen when a job configuration is updated by the plugin. The `JobSweeper` maintains a map of jobs that it has "swept" (schedule/deschedule) which occurs every time a plugin indexes/updates another job to it's registered job index. 

The reason why the `JobDocVersion` is included as part of the request to extensions is because it is one of the parameters of `ScheduledJobParser.parse(XcontentParser, docId, jobDocVersion)`.

While the Anomaly Detection implementation of the `ScheduledJobParser` utilizes only the provided `xContentParser`, I felt that it would be necessary to add support to transport both the `docId` and also the `JobDocVersion` as well, in case an extension required this information.

Reply:
For instance, ISM requires the `jobDocVersion` [here](https://github.com/opensearch-project/index-management/blob/76d530fe1c4a46d0dab8a179557f30b31bcc2e91/src/main/kotlin/org/opensearch/indexmanagement/IndexManagementPlugin.kt#LL237C46-L237C60) within their `ScheduledJobParser` implementation